### DISCUSSION TITLE
Security - minimal permisions

### DISCUSSION THREAD

**Discussion by BrubiSQL** [trimble-oss/dba-dash#176] [General]
Suggestion for edit of this doc: https://github.com/trimble-oss/dba-dash/blob/main/Docs/Security.md

To allow DBA Dash to read Jobs, and stuff relating to it, you need to add 

```
USE [msdb]
GO
ALTER ROLE [SQLAgentReaderRole] ADD MEMBER [DBAdashUser]
GO
```

**Comment by DavidWiseman** [comment]
The security doc includes adding to the db_datareader role.  I might be missing something that needs SQLAgentReaderRole though.  I'll check it out.

**Reply by BrubiSQL** [reply]
My DBA Dash user had msdb reader role added, but he was unable to load SQL Agent jobs. After I've added SQLAgentReaderRole, DBA Dash happily ingested all my Jobs data.

**Comment by DavidWiseman** [comment]
I can confirm that SQLAgentReaderRole is required.  The app behavior is:

Job stats are collected but job details including the name of the job are not.  
No errors occur with the collection - the data just isn't returned.

This data is collected via SMO.  Internally this is using dbo.sysjobs_view which doesn't return any rows without membership of SQLAgentReaderRole.  Interestingly if you query dbo.sysjobs instead you get similar data returned without needing to be a member of SQLAgentReaderRole.

I'll update the documentation to add the SQLAgentReaderRole role as a requirement. #182

**Comment by DavidWiseman** [comment]
Docs have now been updated
